audit_lambda.py

```diff
@@ -33,48 +37,42 @@
     storage.set_options(storage_options)
 
 
+def get_adv_status(repo):
+    response = github_rest_client.get(
+        f"/repos/{repo.owner.login}/{repo.name}/vulnerability-alerts"
+    )
+    alerts_enabled = response.status_code == 204
+    vulnerable = repo.vulnerabilityAlerts.edges
+
+    if vulnerable:
+        repo.status = "vulnerable"
+    elif alerts_enabled:
+        repo.status = "clean"
+    else:
+        repo.status = "disabled"
+
+    # append status to repo in original repositories file
+    repo.securityAdvisoriesEnabledStatus = alerts_enabled
+
+    return repo
+
+
 def update_github_advisories_status():
+    log.debug("update_github_advisories_status")
     today = datetime.date.today().isoformat()
     current = get_current_audit()
 
     by_alert_status = defaultdict(list)
 
     today_repositories = storage.read_json(f"{today}/data/repositories.json")
     current_repositories = storage.read_json(f"{current}/data/repositories.json")
-    for today_repo in today_repositories["public"]:
-        new_repo = True
-        update_status = False
-        found_repo = None
-        for current_repo in current_repositories["public"]:
-            if today_repo.name == current_repo.name:
-                found_repo = current_repo
-                if not current_repo.securityAdvisoriesEnabledStatus:
-                    update_status = True
-                    new_repo = False
-
-        if new_repo | update_status:
-
-            response = github_rest_client.get(
-                f"/repos/{today_repo.owner.login}/{today_repo.name}/vulnerability-alerts"
-            )
-            alerts_enabled = response.status_code == 204
-            vulnerable = today_repo.vulnerabilityAlerts.edges
-
-            if vulnerable:
-                status = "vulnerable"
-            elif alerts_enabled:
-                status = "clean"
-            else:
-                status = "disabled"
 
-            # append status to repo in original repositories file
-            today_repo.securityAdvisoriesEnabledStatus = alerts_enabled
+    for state in today_repositories.keys():
+        log.debug(f"{state} repos: {len(today_repositories[state])}")
 
-            by_alert_status[status].append(today_repo)
-            time.sleep(0.1)
-        elif found_repo:
-            alerts_enabled = found_repo.securityAdvisoriesEnabledStatus
-            today_repo.securityAdvisoriesEnabledStatus = alerts_enabled
+        with ThreadPoolExecutor(max_workers=20) as executor:
+            for r in executor.map(get_adv_status, today_repositories[state]):
+                by_alert_status[r.status].append(r)
```

Comment:
this is :fire: 

Shameless plug for my parallel map pattern: https://gist.github.com/0atman/31d85fbad4e1cf68b218267c8b57d5ae

```python
from multiprocessing import Pool

def pmap(f, collection, size=10):
    """
    Applies `f` in parallel over `collection`.
    Pool `size` has a sensible default of 10.
    """
    with Pool(size) as p:
        return p.map(f, collection)
```